spec/abci/apps.md

```diff
@@ -315,6 +315,18 @@ txs included in a proposed block.
 Must have `MaxGas >= -1`.
 If `MaxGas == -1`, no limit is enforced.
 
+### BlockParams.RecheckTx
+
+This indicates whether all nodes in the network should perform a `CheckTx` on all
+transactions in the mempool directly *after* the execution of every block i.e.
+whenever a new application state is created. This is often useful for garbage
+collection.
+
+The change will come into effect immediately after `FinalizeBlock` has been
```

Comment:
Is this different from other consensus parameters in this regard?

Reply:
Hmm not really. Consensus params are in effect from the very next height. We do recheck tx right after `FinalizeBlock` and before we start proposing the next block so it's slightly more immediate but not really. I thought it would be good to be explicit nonetheless